framework/src/node/forger/forger.ts

```diff
@@ -91,6 +90,35 @@ interface CreateBlockInput {
 }
 
 const BLOCK_VERSION = 2;
+const forgerInfoMissing = (
+	previouslyForgedMap: dataStructures.BufferMap<ForgedInfo>,
+	forgerAddress: Buffer,
+	{ height, maxHeightPreviouslyForged, maxHeightPrevoted }: ForgedInfo,
+	overwrite?: boolean,
+) =>
+	!overwrite &&
+	!previouslyForgedMap.has(forgerAddress) &&
+	maxHeightPreviouslyForged !== 0 &&
```

Comment:
I don't understand why we check that the inputs are non-zero here (this check and the following two checks).